packages/mobile/src/send/reducers.ts

```diff
@@ -1,18 +1,50 @@
+import BigNumber from 'bignumber.js'
+import { TokenTransactionType } from 'src/apollo/types'
 import { areRecipientsEquivalent, Recipient } from 'src/recipients/recipient'
 import { getRehydratePayload, REHYDRATE, RehydrateAction } from 'src/redux/persist-helper'
+import { RootState } from 'src/redux/reducers'
 import { Actions, ActionTypes } from 'src/send/actions'
 
 // Sets the limit of recent recipients we want to store
 const RECENT_RECIPIENTS_TO_STORE = 8
 
+export interface TransactionData {
+  recipient: Recipient
+  amount: BigNumber
+  reason: string
+  type: TokenTransactionType
+  firebasePendingRequestUid?: string | null
+}
+
+export interface ConfirmationInput {
+  recipient: Recipient
+  amount: BigNumber
+  reason: string
+  recipientAddress: string | null | undefined
+  type: TokenTransactionType
+  firebasePendingRequestUid?: string | null
+}
+
+export interface SecureSendPhoneNumberMapping {
+  [e164Number: string]: {
+    address: string | undefined
+    addressValidationRequired: boolean
+    fullValidationRequired: boolean
```

Comment:
What's the difference between these two fields? Could one be implied by the fact that address may be undefined, and thereby save on storage space?

Reply:
I believe we need both. `address` is `undefined` until validation is successful so its value can't be used as a leading indicator.

true + true  = full address validation needed
true + false = last 4 digits of address validation needed
false + false = no validation needed

Reply:
On second thought, I could make `fullValidiationRequired` optional if the space concern warrants it. What do you think?

Reply:
How about an enum representing the diff possible states instead of two flags?

Reply:
Yeah, that's probably a better solution. Will do.